### PR TITLE
Build script for fork of distillpub/template

### PR DESCRIPTION
**Pull request by kevinykuo**
This script
1. Clones the https://github.com/rstudio/template repo into a temp directory, 
2. builds the .js artifacts, 
3. applies the changes https://github.com/rstudio/radix/commits/master/inst/www/distill/template.v2.js, 
4. then copies the `template.v2.js` file into `inst/www/distill`.

This allows us to maintain our own fork of the distill template and having the changes reflected in radix, which allows bugs to be fixed sooner than the upstream repo.

Note that there is currently no automated mechanism to protect malicious changes (diffs in the minified js file are hard to see.) We could consider adding a test to build the artifact on travis and compare the checksum against what's in `inst/`.

### CODE REVIEW

File: configure_template.R [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+library(git2r)
+library(stringr)
+library(readr)
+path <- file.path(tempfile(pattern = "git2r-"), "template")
+dir.create(path, recursive = TRUE)
+repo <- clone("https://github.com/rstudio/template", branch = "radix", path)
+system2("npm", args = c("--prefix", path, "install"))
+system2("npm", args = c("--prefix", path, "run", "build"))
+read_lines(file.path(path, "dist", "template.v2.js")) %>%
+  str_replace(fixed("/[a-z0-9_]+(?=\\()/i"), "/[a-z\\.0-9_]+(?=\\()/i") %>%
```
Comment (jjallaire) [inline configure_template.R]: Do you feel confident that these regexes will only apply the changes we did before? (i.e. could they leak to other parts of the script). If you are a bit uncertain is there a way to scope the change down to make it safer?

Comment (jjallaire) [reply configure_template.R]: LMK what you think about the regexes (save for that it looks like we are ready to merge).

**Comment by jjallaire** [comment]
Okay sounds good!

**Comment by kevinykuo** [comment]
Yeah, not sure what I was seeing before, but this is b38b357fcc130d626ae23d6efb7330d0da8f4050 vs 7beb79ef9b287e6a3c0abfc88967f2d347afa7b4: https://www.diffchecker.com/5STvA5nt.

**Comment by jjallaire** [comment]
Okay, that LGTM. Are we ready to merge?

**Comment by jjallaire** [comment]
Actually, if you could also include the modified distillpub template.js file in the commit I can check it's behavior against existing Distill sites.

**Comment by jjallaire** [comment]
I get the following JS exception when loading a page with the updated template:

screen shot 2019-01-03 at 12 23 31 pm

**Comment by kevinykuo** [comment]
Should be OK now.

**Comment by jjallaire** [comment]
Great, thanks! I tested out my 2 larger Radix sites and all seems to be well.

**Pull request merged**